internal/provider/resource_device.go

```diff
@@ -0,0 +1,215 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/paultyng/go-unifi/unifi"
+)
+
+func resourceDevice() *schema.Resource {
+	return &schema.Resource{
+		Description: "`unifi_device` manages a device of the network.\n\n" +
+			"Devices are adopted by the controller, so it may not be possible " +
+			"for this resource to be created through terraform.",
+
+		//Create: resourceDeviceCreate,
+		Read:   resourceDeviceRead,
+		Update: resourceDeviceUpdate,
+		Delete: resourceDeviceDelete,
+		Importer: &schema.ResourceImporter{
+			State: importSiteAndID,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The ID of the device.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"site": {
+				Description: "The name of the site to associate the device with.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"mac": {
+				Description:      "The MAC address of the device.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: macDiffSuppressFunc,
+				ValidateFunc:     validation.StringMatch(macAddressRegexp, "Mac address is invalid"),
+			},
+			"name": {
+				Description: "The name of the device.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"disabled": {
+				Description: "Specifies whether this device should be disabled.",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"port_overrides": {
+				Description: "Settings overrides for specific switch ports.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"port_idx": {
+							Description: "Switch port number.",
+							Type:        schema.TypeInt,
+							Required:    true,
+						},
+						"name": {
+							Description: "Human-readable name of the port.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"port_profile_id": {
+							Description: "ID of the Port Profile used on this port.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceDeviceCreate(d *schema.ResourceData, meta interface{}) error {
+	return &unifi.NotFoundError{}
```

Comment:
Is the thought this is "import only" since you would need to adopt it to create anyway?

Reply:
Yes, that was the idea, but I haven't check if I can not have Create method.
In any case, this should probably be some other error type, but I didn't know what type would be good here.

Reply:
Looks like not having Create method makes terraform use some default one / call Update instead, so I made create and delete return a more descriptive error.